didx-xyz/yoma-mobile
# DON'T REVIEW YET - YOMA-225 - feat(Login): setup redux boilerplate

## NOTE: Seeing some weirdness with the pr.
Not sure what's going on with the PR - a bunch of changes that I didn't make this time round - so want to figure out what that's about before we continue - will probably create a fresh branch with this in in the morning. 

![](https://media.giphy.com/media/13py6c5BSnBkic/giphy.gif)
## Scope [YOMA-225](https://yomaworld.atlassian.net/browse/YOMA-225)
 - Initial work getting a first example of redux into the project for login

## Work Done
- adding proper actions, middleware, reducers for login
- updating Login view to handle reducer and dispatch actions through container wrapper.
- moved Login to module root
- added some types, etc where required.

## Steps to Test
- make sure you have React Native Debugger or some other RN console running
- open the app and try to login
- see that the success and failure states are handled with notifications
- see that the success data is saved into the redux state.
    
## Screenshots
<img width="1469" alt="Screenshot 2021-04-25 at 23 44 12" src="https://user-images.githubusercontent.com/1424561/116010626-6ba3e700-a620-11eb-9cec-fd54a4e2fe91.png">
<img width="1346" alt="Screenshot 2021-04-25 at 23 44 21" src="https://user-images.githubusercontent.com/1424561/116010633-765e7c00-a620-11eb-9cd3-a874c07597ed.png">
